fvp.mk

```diff
@@ -146,19 +155,72 @@ filelist-tee: filelist-tee-common
 update_rootfs: update_rootfs-common
 
 ################################################################################
+# grub
+################################################################################
+grub-flags := CC="$(CCACHE)gcc" \
+	TARGET_CC="$(AARCH64_CROSS_COMPILE)gcc" \
+	TARGET_OBJCOPY="$(AARCH64_CROSS_COMPILE)objcopy" \
+	TARGET_NM="$(AARCH64_CROSS_COMPILE)nm" \
+	TARGET_RANLIB="$(AARCH64_CROSS_COMPILE)ranlib" \
+	TARGET_STRIP="$(AARCH64_CROSS_COMPILE)strip"
+
+GRUB_MODULES += boot chain configfile echo efinet eval ext2 fat font gettext \
+		gfxterm gzio help linux loadenv lsefi normal part_gpt \
+		part_msdos read regexp search search_fs_file search_fs_uuid \
+		search_label terminal terminfo test tftp time
+
+$(GRUB_PATH)/configure: $(GRUB_PATH)/configure.ac
+	cd $(GRUB_PATH) && ./autogen.sh
+
+$(GRUB_PATH)/Makefile: $(GRUB_PATH)/configure
+	cd $(GRUB_PATH) && ./configure --target=aarch64 --enable-boot-time $(grub-flags)
+
+.PHONY: grub
+grub: $(GRUB_PATH)/Makefile | $(OUT_PATH)
```

Comment:
What does the `|` do?

Reply:
That will create the out path only when necessary (i.e., avoid updating it). It's the so called [order-only-prerequisites](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html). If you aren't using that, then it will always run the `$(OUT_PATH)` target which will update the timestamp on that folder and therefore could/will trigger unnecessary (re-)builds. [Here](https://www.cmcrossroads.com/article/making-directories-gnu-make) is a pretty good example and explanation of this.

Reply:
Very good to know thanks!